src/test/java/com/uber/cadence/workflow/WorkflowTest.java

```diff
@@ -951,6 +951,34 @@ public void testSyncUntypedAndStackTrace() {
     assertEquals("activity10", result);
   }
 
+  @Test
+  public void testUntypedAsyncStart() {
+    startWorkerFor(TestSyncWorkflowImpl.class);
+    WorkflowStub startWorkflowStub =
+        workflowClient.newUntypedWorkflowStub(
+            "TestWorkflow1::execute", newWorkflowOptionsBuilder(taskList).build());
+    CompletableFuture<WorkflowExecution> future = startWorkflowStub.startAsync(taskList);
+
+    try {
+      WorkflowExecution execution = future.get();
+      WorkflowStub workflowStub =
```

Comment:
Set workflowExecution on completion in WorkflowStubImpl.startAsyncWithOptions, so we can reuse the existing stub.

Reply:
make sense. Done